discretisedfield/mesh.py

```diff
@@ -187,22 +182,42 @@ def __init__(
         attributes={"unit": "m"},
     ):
         # TODO NO MUTABLE DEFAULT
-        if region is not None and p1 is None and p2 is None:
-            self.region = region
+        if region is None and p1 is None and p2 is None:
+            raise ValueError("p1, p2, and region cannot be None simultaneously.")
```

Comment:
Do we need two separate checks for "all None", which would anyway fail if e.g. only region and p1 are not None, and the `else` (line 193)?